### ISSUE TITLE
When i add the handlers method to the react quill, the edit box always auto blur when i input a letter.

### ISSUE THREAD

**Issue by z-lionel** [zenoamaro/react-quill#688]
When i add the handlers method to the react quill, the edit box always auto blur when i input a letter.
how can i fix it?

I want to add custom properties for the link like this:
 < a href=" " data-custom="custom properties">data</ a>

it means i have to custome the handlers link?
because i can't find any api to do this thing.

**Comment by vidimitrov** [comment]
I also experience the same and I was wondering if I am doing something wrong, but it looks like it comes from the ReactQuill component...

@zenoamaro any ideas?

**Comment by SooJungChae** [comment]
I have same issue get an error `Error: The given range isn't in document` in browser console.
And fix this wrapping with useMemo function.
reference: https://github.com/quilljs/quill/issues/1940

```js
  const modules = useMemo(() => {
      toolbar: {
        handlers: {
          image: imageHandler,
        },
        container: [
          [{ header: [1, 2, false] }],
          ['bold', 'italic', 'underline', 'strike', 'blockquote'],
          [{ list: 'ordered' }, { list: 'bullet' }, { indent: '-1' }, { indent: '+1' }],
          ['link', 'image'],
          ['clean'],
        ],
      },
      clipboard: {
        matchVisual: false,
      },
    };
  }, []);
```

```js
<ReactQuill
      theme="snow"
      id={'quill'}
      style={{ marginBottom: '10px' }}
      modules={modules}
      value={content}
      onChange={onChangeContent}
      ref={quillRef}
    />
```

**Comment by bengotow** [comment]
Yep running in to this as well - it seems typing the first character into an empty ReactQuill editor with the value/onChange set to use strings (vs deltas) causes it to blur. Trying to work around this, will let y'all know if I come up with anything!

**Comment by joeyparis** [comment]
Also experiencing the same problem will let you know if I find anything out too!

**Comment by codefist** [comment]
my workaround
![Screenshot from 2021-04-09 13-17-03](https://user-images.githubusercontent.com/16125676/114217298-fa80e600-9935-11eb-9b2d-a32f6193b08f.png)

**Comment by dieng01** [comment]
I have the same problem:
I using this [library](https://github.com/xnimorz/use-debounce) (use-debounce) to resolve it: 

`<ReactQuill ref={quill} defaultValue={""} onChange={(e) => debounced(e)} modules={modules} />`

`const debounced = useDebouncedCallback((value: string) => {
		setEditorHtml(value)
}, 500)`

**Comment by pavi2410** [comment]
Same issue. https://github.com/zenoamaro/react-quill/issues/688#issuecomment-793441268 worked for me

**Comment by SimonProper** [comment]
Using useMemo worked for me as well, however the brackets and parentheses in https://github.com/zenoamaro/react-quill/issues/688#issuecomment-793441268 is not correct. 
This is what worked for me:

```js
const modules = useMemo(() => ({
      toolbar: {
        handlers: { image: imageHandler,},
        container: [
          [{ header: [1, 2, false] }],
          ["bold", "italic", "underline", "strike", "blockquote"],
          [{ list: "ordered" }, { list: "bullet" }, { indent: "-1" }, { indent: "+1" }],
          ["link", "image"],
          ["clean"],
        ],
      },
      clipboard: {
        matchVisual: false,
      },
    }),[]);
```

**Comment by tsingroo** [comment]
> Using useMemo worked for me as well, however the brackets and parentheses in [#688 (comment)](https://github.com/zenoamaro/react-quill/issues/688#issuecomment-793441268) is not correct. This is what worked for me:
> 
> ```js
> const modules = useMemo(() => ({
>       toolbar: {
>         handlers: { image: imageHandler,},
>         container: [
>           [{ header: [1, 2, false] }],
>           ["bold", "italic", "underline", "strike", "blockquote"],
>           [{ list: "ordered" }, { list: "bullet" }, { indent: "-1" }, { indent: "+1" }],
>           ["link", "image"],
>           ["clean"],
>         ],
>       },
>       clipboard: {
>         matchVisual: false,
>       },
>     }),[]);
> ```

worked for me

**Comment by keinn51** [comment]
I love useMemo
